openhands/sdk/conversation/secrets_manager.py

```diff
@@ -0,0 +1,104 @@
+"""Secrets manager for handling sensitive data in conversations."""
+
+import re
+from typing import Callable, Dict
+
+from openhands.sdk.logger import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class SecretsManager:
+    """Manages secrets and injects them into bash commands when needed.
+
+    The secrets manager stores a mapping of secret keys to callable functions
+    that retrieve the actual secret values. When a bash command is about to be
+    executed, it scans the command for any secret keys and injects the corresponding
+    environment variables.
```

Comment:
if we are not export the env var, will the agent have issue accessing these secret (eg in a python script), unless the agent explicitly export them before running the script?

If the agent is able to export them to shell, why not just export them in the first place to simplify things a bit? I think this can be just used to strip out sensitive values from the Events

Reply:
Yep exactly 

OH didnt do a good first shot at this I'll be iterating on this PR during my downtimes

Reply:
@openhands address this feedback - we should be exporting secrets and not modifying the action events themselves

don't modify or run any unit tests for now, just focus on redoing this